### ISSUE TITLE
SiloHostBuilder not validating GrainCollectionOptions during build

### ISSUE THREAD

**Issue by JoshCollinsMSFT** [dotnet/orleans#5200]
**Repro code:**

```
var siloHostBuilder = new SiloHostBuilder()                    
                    .Configure<GrainCollectionOptions>(options =>
                    {
                        options.CollectionAge = TimeSpan.FromSeconds(30);
                    });

siloHostBuilder.Build();
```

**Expected behavior**
The call to siloHostBuilder.Build() should throw an exception because the specified TimeSpan for options.Collection age is less than 1 minute.

**Actual behavior**
The call to siloHostBuilder.Build() succeeds, however, activations will fail because the TimeSpan is less than 1 minutes.  Moreover, when the runtime throws the exception on the silo when the activation fails, the exception never reaches the client which made the initial request to the cluster, and the client experiences a timeout.

**Comment by xiazen** [comment]
SiloHostBuilder do validate GrainCollectionOptions.CollectionQuantum, but yes it doesn't validate GrainCollectionOptions.CollectionAge, which is currently default to be 2hr. I think we should validate it, I'm just not sure what's the good minimum for it. Can you explain why it should be more than 1 min at least? Instead of 30 sec, or more than 0? 

And another issue I see is, if an activation failed, the client gets an timeout. As to today, I think this is the designed behavior. But I do see we can add more detailed message back to the client to provide more information on why it timed out, whether it is grain failed activation or the actual calling method timed out. Is this sufficient to improve the situation you saw?

**Comment by sergeybykov** [comment]
`CollectionAge` must be longer than `CollectionQuantum` which by default is 1 minute.

**Comment by xiazen** [comment]
Oh I see. yeep we don't do that.

**Comment by xiazen** [comment]
fixed in #5290
